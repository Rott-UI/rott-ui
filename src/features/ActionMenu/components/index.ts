export * from './ActionMenuComponent'
export * from './ActionMenuHeaderComponent'
