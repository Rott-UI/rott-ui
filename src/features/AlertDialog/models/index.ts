export * from './alertDialogButtonProps'
export * from './alertDialogContextModel'
export * from './alertDialogModel'
