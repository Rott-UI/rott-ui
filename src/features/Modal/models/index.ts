export * from './modalProps'
export * from './modalContextModel'
