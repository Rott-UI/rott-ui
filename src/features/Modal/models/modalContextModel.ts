import {ModalProps} from './modalProps'

export interface ModalContextModel {
  modals: ModalProps[]
  showModal: (modalToRender: ModalProps) => void
  updateModal: (modalToRender: ModalProps, id?: number) => void
  hideModal: (id?: number) => void
  hasModalById: (id: number) => boolean
  hideAllModal: () => void
}
