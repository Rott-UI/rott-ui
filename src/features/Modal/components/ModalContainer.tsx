/* eslint-disable react-native/no-inline-styles */
// React Imports
import React, {FC} from 'react'

// React Native Imports
import {GestureResponderEvent, ScrollView} from 'react-native'

// Component Imports
import {ModalProps} from '../models'
import {Content} from '../../Content'
import {Pressable} from '../../Pressable'

interface ModalContainerProps extends ModalProps {
  onPress?: (event: GestureResponderEvent) => void
}

export const ModalContentContainer: FC<ModalContainerProps> = ({
  transparent,
  height,
  backgroundColor,
  fullScreen,
  justifyContentCenter,
  alignItemsCenter,
  modalContainerMarginBottom,
  onPress,
  children,
}) => {
  if (transparent) {
    return (
      <Content
        height={height}
        noPadding
        backgroundColor={backgroundColor}
        borderTopStartRadius={fullScreen ? 0 : 24}
        borderTopEndRadius={fullScreen ? 0 : 24}
        justifyContentCenter={justifyContentCenter}
        alignItemsCenter={alignItemsCenter}
        marginBottom={modalContainerMarginBottom}>
        <ScrollView
          keyboardShouldPersistTaps='handled'
          showsVerticalScrollIndicator={false}
          showsHorizontalScrollIndicator={false}
          scrollEnabled={false}
          contentContainerStyle={{
            flex: fullScreen ? 0 : 1,
            justifyContent: justifyContentCenter ? 'center' : undefined,
            alignItems: alignItemsCenter ? 'center' : undefined,
          }}>
          {children}
        </ScrollView>
      </Content>
    )
  } else {
    return (
      <Pressable
        onPress={onPress}
        height={height}
        backgroundColor={backgroundColor}
        borderTopStartRadius={fullScreen ? 0 : 24}
        borderTopEndRadius={fullScreen ? 0 : 24}
        justifyContentCenter={justifyContentCenter}
        alignItemsCenter={alignItemsCenter}
        marginBottom={modalContainerMarginBottom}>
        <ScrollView
          keyboardShouldPersistTaps='handled'
          showsVerticalScrollIndicator={false}
          showsHorizontalScrollIndicator={false}
          scrollEnabled={false}
          contentContainerStyle={{
            flex: fullScreen ? 0 : 1,
            justifyContent: justifyContentCenter ? 'center' : undefined,
            alignItems: alignItemsCenter ? 'center' : undefined,
          }}>
          {children}
        </ScrollView>
      </Pressable>
    )
  }
}
