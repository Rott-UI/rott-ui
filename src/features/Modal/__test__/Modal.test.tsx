// React Imports
import React from 'react'

// Component Imports
import {Content} from '../../Content'
import {Item} from '../../Item'
import {Label} from '../../Label'
import {ModalComponent} from '../components'
import {ModalProps} from '../models'

// Util and Lib Imports
import {act, fireEvent, render, waitFor} from '../../../utils'

describe('Modal -> Custom Component', () => {
  const onCloseMock = jest.fn()

  const testId = {
    modalTestId: 'modal-test-id',
    headerTestId: 'modal-header-test-id',
    headerCloseButtonTestId: 'header-right-pressable-test-id',
    headerCloseIconTestId: 'header-right-icon-test-id',
    outsideTapAreaTestId: 'outside-tap-area-test-id',
    slideToCloseTestId: 'slide-to-close-button-test-id',
  }

  const dummyDataFullScreen: ModalProps = {
    testID: testId.modalTestId,
    header: {
      title: 'Test',
      logo: 'PTTBANK_BLACK_COLORED',
    },
    slideToClose: true,
    visible: true,
    fullScreen: true,
    closeButton: true,
    onClose: onCloseMock,
  }

  const dummyData: ModalProps = {
    testID: testId.modalTestId,
    header: {
      title: 'Test',
      logo: 'PTTBANK_BLACK_COLORED',
    },
    visible: true,
    height: 50,
    fullScreen: false,
    slideToClose: true,
    closeButton: true,
    onClose: onCloseMock,
  }

  beforeAll(() => {
    // TODO: useState mock example
    const setStateMock = jest.fn()
    const useStateMock: any = (useState: any) => [useState, setStateMock]
    jest.spyOn(React, 'useState').mockImplementation(useStateMock)
  })

  it('modal ilk renderlandiginda snapshot ile eşleşmeli', () => {
    const renderedModal = render(
      <ModalComponent {...dummyData}>
        <Item>
          <Label>{'Test'}</Label>
        </Item>
      </ModalComponent>
    )

    expect(renderedModal).toMatchSnapshot()
  })

  it('modal visible değilken ekranda gözükmemeli', async () => {
    const {queryByText} = render(
      <ModalComponent {...dummyData} visible={false}>
        <Item>
          <Label>Test</Label>
        </Item>
      </ModalComponent>
    )

    await act(() => {
      const testLabelByText = queryByText('Test')

      expect(testLabelByText).not.toBeOnTheScreen()
    })
  })

  it('custom header verildiğinde ekranda custom header olmalı', async () => {
    const {headerTestId} = testId

    const {getByTestId} = render(
      <ModalComponent
        {...dummyData}
        header={
          <Content testID={headerTestId}>
            <Label>{'Test'}</Label>
          </Content>
        }>
        <Item>
          <Label>Test</Label>
        </Item>
      </ModalComponent>
    )

    await act(() => {
      const modalHeader = getByTestId(headerTestId)

      expect(modalHeader).toBeOnTheScreen()
    })
  })

  it('modal kapatma butonu headerda olmalı', async () => {
    const {headerCloseIconTestId} = testId
    const {getByTestId} = render(<ModalComponent {...dummyData} />)

    await act(() => {
      const headerCloseButton = getByTestId(headerCloseIconTestId)

      expect(headerCloseButton).toBeOnTheScreen()
    })
  })

  it('modal header child elementi ekranda olmalı', async () => {
    const {getByTestId} = render(
      <ModalComponent
        {...dummyData}
        header={{
          title: 'Test',
          logo: 'PTTBANK_BLACK_COLORED',
          children: <Label testID={'header-children-test-id'}>{'Test'}</Label>,
        }}>
        <Item>
          <Label>Test</Label>
        </Item>
      </ModalComponent>
    )

    await act(() => {
      const headerChildren = getByTestId('header-children-test-id')

      expect(headerChildren).toBeOnTheScreen()
    })
  })

  it('modal kapatma butonuna tıklandığında modal kapanmalı', async () => {
    const onClosePressableMock = jest.fn()
    const {headerCloseButtonTestId} = testId
    const {getByTestId} = render(<ModalComponent {...dummyData} onClose={onClosePressableMock} />)

    await act(() => {
      const headerCloseButton = getByTestId(headerCloseButtonTestId)
      fireEvent.press(headerCloseButton)
    })

    await waitFor(() => expect(onClosePressableMock).toHaveBeenCalledTimes(1))
  })

  it('modal tam ekran değilken container dışına tıklanırsa modal kapanmalı', async () => {
    const onClosePressableMock = jest.fn()
    const {outsideTapAreaTestId} = testId
    const {getByTestId} = render(<ModalComponent {...dummyData} onClose={onClosePressableMock} />)

    await act(() => {
      const outsideContainer = getByTestId(outsideTapAreaTestId)
      fireEvent.press(outsideContainer)
    })

    await waitFor(() => expect(onClosePressableMock).toHaveBeenCalledTimes(1))
  })

  it('panResponder headerda olmalı', async () => {
    const {slideToCloseTestId} = testId
    const {getByTestId} = render(<ModalComponent {...dummyData} />)

    await act(() => {
      const panResponder = getByTestId(slideToCloseTestId)

      expect(panResponder).toBeOnTheScreen()
    })
  })

  it('panResponder headerda olmamalı', async () => {
    const {slideToCloseTestId} = testId
    const {queryByTestId} = render(<ModalComponent {...dummyData} slideToClose={false} />)

    await act(() => {
      const panResponder = queryByTestId(slideToCloseTestId)

      expect(panResponder).toBeNull()
    })
  })

  it('panResponder header dışında olmalı', async () => {
    const {slideToCloseTestId, headerTestId} = testId
    const {getByTestId} = render(<ModalComponent {...dummyData} />)

    await act(() => {
      const panResponder = getByTestId(slideToCloseTestId)

      // TODO: İç elementleri de bir katman olarak çağırıyor.
      expect(panResponder.parent?.parent?.parent).not.toHaveProp('testID', headerTestId)
    })
  })

  it('modal fullscreen modunda panResponder ekranda olmamalı', async () => {
    const {slideToCloseTestId} = testId
    const {queryByTestId} = render(<ModalComponent {...dummyDataFullScreen} />)

    await act(() => {
      const panResponder = queryByTestId(slideToCloseTestId)

      expect(panResponder).not.toBeOnTheScreen()
    })
  })

  it('modal verilen children elementi ekranda göstermeli', async () => {
    const {getByText} = render(
      <ModalComponent fullScreen visible>
        <Label>{'Test'}</Label>
      </ModalComponent>
    )

    await act(() => {
      const testLabelByText = getByText('Test')

      expect(testLabelByText).toBeOnTheScreen()
    })
  })
})
