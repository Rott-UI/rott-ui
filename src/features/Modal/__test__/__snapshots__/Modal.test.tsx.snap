// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal -> Custom Component modal ilk renderlandiginda snapshot ile eşleşmeli 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  statusBarTranslucent={true}
  style={
    {
      "height": 422,
    }
  }
  testID="modal-test-id"
  transparent={true}
  visible={true}
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    animated={false}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignContent": "center",
        "alignItems": undefined,
        "backgroundColor": "rgba(34, 63, 70, 0.75)",
        "bottom": undefined,
        "flex": 1,
        "flexDirection": "column",
        "flexGrow": undefined,
        "gap": undefined,
        "height": undefined,
        "justifyContent": "center",
        "left": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "maxHeight": undefined,
        "maxWidth": undefined,
        "minHeight": undefined,
        "minWidth": undefined,
        "overflow": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "paddingVertical": undefined,
        "position": undefined,
        "right": undefined,
        "top": undefined,
        "width": undefined,
        "zIndex": undefined,
      }
    }
    testID="outside-tap-area-test-id"
  />
  <View
    backgroundColor="rgba(34, 63, 70, 0.75)"
    style={
      {
        "alignItems": undefined,
        "backgroundColor": "rgba(34, 63, 70, 0.75)",
        "bottom": undefined,
        "flex": undefined,
        "flexDirection": "row",
        "flexGrow": undefined,
        "gap": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "left": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "maxHeight": undefined,
        "maxWidth": undefined,
        "minHeight": undefined,
        "minWidth": undefined,
        "overflow": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "paddingVertical": undefined,
        "position": undefined,
        "right": undefined,
        "top": undefined,
        "width": undefined,
        "zIndex": undefined,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "elevation": 5,
          "flex": 1,
          "flexDirection": "row",
          "shadowColor": "rgba(6, 14, 16, 0.30)",
          "shadowOffset": {
            "height": -8,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 35,
          "transform": [
            {
              "translateY": 0,
            },
          ],
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        animated={false}
        backgroundColor="rgba(255, 255, 255, 1)"
        borderTopEndRadius={24}
        borderTopStartRadius={24}
        collapsable={false}
        focusable={true}
        height={422}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": undefined,
            "backgroundColor": "rgba(255, 255, 255, 1)",
            "bottom": undefined,
            "flex": 1,
            "flexDirection": "column",
            "flexGrow": undefined,
            "gap": undefined,
            "height": 812,
            "justifyContent": undefined,
            "left": undefined,
            "marginBottom": undefined,
            "marginLeft": undefined,
            "marginRight": undefined,
            "marginTop": undefined,
            "maxHeight": 812,
            "maxWidth": undefined,
            "minHeight": undefined,
            "minWidth": undefined,
            "overflow": undefined,
            "paddingBottom": undefined,
            "paddingHorizontal": undefined,
            "paddingLeft": undefined,
            "paddingRight": undefined,
            "paddingTop": undefined,
            "paddingVertical": undefined,
            "position": undefined,
            "right": undefined,
            "top": undefined,
            "width": undefined,
            "zIndex": undefined,
          }
        }
        testID="pressable-test-id"
      >
        <RCTScrollView
          contentContainerStyle={
            {
              "alignItems": undefined,
              "flex": 1,
              "justifyContent": undefined,
            }
          }
          keyboardShouldPersistTaps="handled"
          scrollEnabled={false}
          showsHorizontalScrollIndicator={false}
          showsVerticalScrollIndicator={false}
        >
          <View>
            <View
              alignItemsCenter={true}
              backgroundColor="rgba(61, 88, 94, 1)"
              borderTopEndRadius={24}
              borderTopStartRadius={24}
              justifyContentCenter={true}
              onMoveShouldSetResponder={[Function]}
              onMoveShouldSetResponderCapture={[Function]}
              onResponderEnd={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderReject={[Function]}
              onResponderRelease={[Function]}
              onResponderStart={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              onStartShouldSetResponderCapture={[Function]}
              paddingTop={8}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "rgba(61, 88, 94, 1)",
                  "bottom": undefined,
                  "flex": undefined,
                  "flexDirection": "column",
                  "flexGrow": undefined,
                  "gap": undefined,
                  "height": undefined,
                  "justifyContent": "center",
                  "left": undefined,
                  "marginBottom": undefined,
                  "marginLeft": undefined,
                  "marginRight": undefined,
                  "marginTop": undefined,
                  "maxHeight": undefined,
                  "maxWidth": undefined,
                  "minHeight": undefined,
                  "minWidth": undefined,
                  "overflow": undefined,
                  "paddingBottom": undefined,
                  "paddingHorizontal": undefined,
                  "paddingLeft": undefined,
                  "paddingRight": undefined,
                  "paddingTop": 16,
                  "paddingVertical": undefined,
                  "position": undefined,
                  "right": undefined,
                  "top": undefined,
                  "width": "100%",
                  "zIndex": undefined,
                }
              }
            >
              <View
                alignItemsCenter={true}
                backgroundColor="rgba(61, 88, 94, 1)"
                borderTopEndRadius={24}
                borderTopStartRadius={24}
                justifyContentCenter={true}
                onMoveShouldSetResponder={[Function]}
                onMoveShouldSetResponderCapture={[Function]}
                onResponderEnd={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderReject={[Function]}
                onResponderRelease={[Function]}
                onResponderStart={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                onStartShouldSetResponderCapture={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "rgba(61, 88, 94, 1)",
                    "bottom": undefined,
                    "flex": undefined,
                    "flexDirection": "column",
                    "flexGrow": undefined,
                    "gap": undefined,
                    "height": undefined,
                    "justifyContent": "center",
                    "left": undefined,
                    "marginBottom": undefined,
                    "marginLeft": undefined,
                    "marginRight": undefined,
                    "marginTop": undefined,
                    "maxHeight": undefined,
                    "maxWidth": undefined,
                    "minHeight": undefined,
                    "minWidth": undefined,
                    "overflow": undefined,
                    "paddingBottom": undefined,
                    "paddingHorizontal": undefined,
                    "paddingLeft": undefined,
                    "paddingRight": undefined,
                    "paddingTop": undefined,
                    "paddingVertical": undefined,
                    "position": undefined,
                    "right": undefined,
                    "top": undefined,
                    "width": "100%",
                    "zIndex": undefined,
                  }
                }
              >
                <View
                  style={
                    {
                      "alignItems": undefined,
                      "backgroundColor": undefined,
                      "bottom": undefined,
                      "flex": undefined,
                      "flexDirection": "column",
                      "flexGrow": undefined,
                      "gap": undefined,
                      "height": undefined,
                      "justifyContent": undefined,
                      "left": undefined,
                      "marginBottom": undefined,
                      "marginLeft": undefined,
                      "marginRight": undefined,
                      "marginTop": undefined,
                      "maxHeight": undefined,
                      "maxWidth": undefined,
                      "minHeight": undefined,
                      "minWidth": undefined,
                      "overflow": undefined,
                      "paddingBottom": undefined,
                      "paddingHorizontal": undefined,
                      "paddingLeft": undefined,
                      "paddingRight": undefined,
                      "paddingTop": undefined,
                      "paddingVertical": undefined,
                      "position": undefined,
                      "right": undefined,
                      "top": undefined,
                      "width": undefined,
                      "zIndex": undefined,
                    }
                  }
                  testID="slide-to-close-button-test-id"
                >
                  <SvgMock
                    fill="transparent"
                    height={77}
                    stroke="rgba(234, 239, 240, 0.15)"
                    strokeLinecap="round"
                    strokeLinejoin="round"
                    strokeWidth={1}
                    width={77}
                  />
                </View>
              </View>
              <View
                minHeight={56}
                paddingVertical={0}
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": undefined,
                    "borderBottomEndRadius": undefined,
                    "borderBottomStartRadius": undefined,
                    "borderRadius": undefined,
                    "borderTopEndRadius": undefined,
                    "borderTopStartRadius": undefined,
                    "bottom": undefined,
                    "flex": undefined,
                    "flexDirection": "column",
                    "flexGrow": undefined,
                    "gap": undefined,
                    "height": undefined,
                    "justifyContent": "center",
                    "left": undefined,
                    "marginBottom": undefined,
                    "marginLeft": undefined,
                    "marginRight": undefined,
                    "marginTop": undefined,
                    "maxHeight": undefined,
                    "maxWidth": undefined,
                    "minHeight": 108,
                    "minWidth": undefined,
                    "overflow": undefined,
                    "paddingBottom": undefined,
                    "paddingHorizontal": 46,
                    "paddingLeft": undefined,
                    "paddingRight": undefined,
                    "paddingTop": undefined,
                    "paddingVertical": 0,
                    "position": undefined,
                    "right": undefined,
                    "top": undefined,
                    "width": undefined,
                    "zIndex": undefined,
                  }
                }
                testID="modal-header-test-id"
              >
                <View
                  height={56}
                  style={
                    {
                      "alignItems": undefined,
                      "backgroundColor": undefined,
                      "bottom": undefined,
                      "flex": undefined,
                      "flexDirection": "row",
                      "flexGrow": undefined,
                      "gap": undefined,
                      "height": 108,
                      "justifyContent": undefined,
                      "left": undefined,
                      "marginBottom": undefined,
                      "marginLeft": undefined,
                      "marginRight": undefined,
                      "marginTop": undefined,
                      "maxHeight": 108,
                      "maxWidth": undefined,
                      "minHeight": undefined,
                      "minWidth": undefined,
                      "overflow": undefined,
                      "paddingBottom": undefined,
                      "paddingHorizontal": undefined,
                      "paddingLeft": undefined,
                      "paddingRight": undefined,
                      "paddingTop": undefined,
                      "paddingVertical": undefined,
                      "position": undefined,
                      "right": undefined,
                      "top": undefined,
                      "width": undefined,
                      "zIndex": undefined,
                    }
                  }
                >
                  <View
                    flexShrink={1}
                    height={56}
                    justifyContentCenter={true}
                    minWidth={56}
                    relative={true}
                    style={
                      {
                        "alignItems": undefined,
                        "backgroundColor": undefined,
                        "bottom": undefined,
                        "flex": undefined,
                        "flexDirection": "column",
                        "flexGrow": undefined,
                        "gap": undefined,
                        "height": 108,
                        "justifyContent": "center",
                        "left": undefined,
                        "marginBottom": undefined,
                        "marginLeft": undefined,
                        "marginRight": undefined,
                        "marginTop": undefined,
                        "maxHeight": 108,
                        "maxWidth": undefined,
                        "minHeight": undefined,
                        "minWidth": 108,
                        "overflow": undefined,
                        "paddingBottom": undefined,
                        "paddingHorizontal": undefined,
                        "paddingLeft": undefined,
                        "paddingRight": undefined,
                        "paddingTop": undefined,
                        "paddingVertical": undefined,
                        "position": "relative",
                        "right": undefined,
                        "top": undefined,
                        "width": undefined,
                        "zIndex": undefined,
                      }
                    }
                  >
                    <View
                      accessibilityState={
                        {
                          "busy": undefined,
                          "checked": undefined,
                          "disabled": undefined,
                          "expanded": undefined,
                          "selected": undefined,
                        }
                      }
                      accessibilityValue={
                        {
                          "max": undefined,
                          "min": undefined,
                          "now": undefined,
                          "text": undefined,
                        }
                      }
                      accessible={true}
                      alignItemsCenter={false}
                      alignItemsFlexStart={true}
                      animated={false}
                      collapsable={false}
                      focusable={true}
                      height={56}
                      justifyContentCenter={true}
                      minWidth={56}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        {
                          "alignItems": "flex-start",
                          "backgroundColor": undefined,
                          "bottom": undefined,
                          "flex": 1,
                          "flexDirection": "column",
                          "flexGrow": undefined,
                          "gap": undefined,
                          "height": 108,
                          "justifyContent": "center",
                          "left": undefined,
                          "marginBottom": undefined,
                          "marginLeft": undefined,
                          "marginRight": undefined,
                          "marginTop": undefined,
                          "maxHeight": 108,
                          "maxWidth": undefined,
                          "minHeight": undefined,
                          "minWidth": 108,
                          "overflow": undefined,
                          "paddingBottom": undefined,
                          "paddingHorizontal": undefined,
                          "paddingLeft": undefined,
                          "paddingRight": undefined,
                          "paddingTop": undefined,
                          "paddingVertical": undefined,
                          "position": undefined,
                          "right": undefined,
                          "top": undefined,
                          "width": 108,
                          "zIndex": undefined,
                        }
                      }
                      testID="header-left-pressable-test-id"
                      width={56}
                    />
                  </View>
                  <View
                    alignItemsCenter={true}
                    flex={1}
                    height={56}
                    justifyContentCenter={true}
                    minWidth={56}
                    relative={true}
                    style={
                      {
                        "alignItems": "center",
                        "backgroundColor": undefined,
                        "bottom": undefined,
                        "flex": 1,
                        "flexDirection": "column",
                        "flexGrow": undefined,
                        "gap": undefined,
                        "height": 108,
                        "justifyContent": "center",
                        "left": undefined,
                        "marginBottom": undefined,
                        "marginLeft": undefined,
                        "marginRight": undefined,
                        "marginTop": undefined,
                        "maxHeight": 108,
                        "maxWidth": undefined,
                        "minHeight": undefined,
                        "minWidth": 108,
                        "overflow": undefined,
                        "paddingBottom": undefined,
                        "paddingHorizontal": undefined,
                        "paddingLeft": undefined,
                        "paddingRight": undefined,
                        "paddingTop": undefined,
                        "paddingVertical": undefined,
                        "position": "relative",
                        "right": undefined,
                        "top": undefined,
                        "width": undefined,
                        "zIndex": undefined,
                      }
                    }
                  >
                    <Image
                      height={24}
                      style={
                        {
                          "backgroundColor": undefined,
                          "borderRadius": 50,
                          "bottom": undefined,
                          "flexGrow": undefined,
                          "gap": undefined,
                          "height": 46,
                          "left": undefined,
                          "marginBottom": undefined,
                          "marginLeft": undefined,
                          "marginRight": undefined,
                          "marginTop": undefined,
                          "maxHeight": 46,
                          "maxWidth": undefined,
                          "minHeight": undefined,
                          "minWidth": undefined,
                          "overflow": undefined,
                          "paddingBottom": undefined,
                          "paddingHorizontal": undefined,
                          "paddingLeft": undefined,
                          "paddingRight": undefined,
                          "paddingTop": undefined,
                          "paddingVertical": undefined,
                          "position": undefined,
                          "resizeMode": "contain",
                          "right": undefined,
                          "top": undefined,
                          "width": 244,
                          "zIndex": undefined,
                        }
                      }
                      testID="header-logo-test-id"
                      width={127}
                    />
                  </View>
                  <View
                    alignItemsFlexEnd={true}
                    flexShrink={1}
                    height={56}
                    justifyContentCenter={true}
                    minWidth={56}
                    relative={true}
                    style={
                      {
                        "alignItems": "flex-end",
                        "backgroundColor": undefined,
                        "bottom": undefined,
                        "flex": undefined,
                        "flexDirection": "column",
                        "flexGrow": undefined,
                        "gap": undefined,
                        "height": 108,
                        "justifyContent": "center",
                        "left": undefined,
                        "marginBottom": undefined,
                        "marginLeft": undefined,
                        "marginRight": undefined,
                        "marginTop": undefined,
                        "maxHeight": 108,
                        "maxWidth": undefined,
                        "minHeight": undefined,
                        "minWidth": 108,
                        "overflow": undefined,
                        "paddingBottom": undefined,
                        "paddingHorizontal": undefined,
                        "paddingLeft": undefined,
                        "paddingRight": undefined,
                        "paddingTop": undefined,
                        "paddingVertical": undefined,
                        "position": "relative",
                        "right": undefined,
                        "top": undefined,
                        "width": undefined,
                        "zIndex": undefined,
                      }
                    }
                  >
                    <View
                      accessibilityState={
                        {
                          "busy": undefined,
                          "checked": undefined,
                          "disabled": undefined,
                          "expanded": undefined,
                          "selected": undefined,
                        }
                      }
                      accessibilityValue={
                        {
                          "max": undefined,
                          "min": undefined,
                          "now": undefined,
                          "text": undefined,
                        }
                      }
                      accessible={true}
                      alignItemsCenter={false}
                      alignItemsFlexEnd={true}
                      alignItemsFlexStart={true}
                      animated={false}
                      collapsable={false}
                      focusable={true}
                      height={56}
                      justifyContentCenter={true}
                      minWidth={56}
                      name="REMOVE_BIG"
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        {
                          "alignItems": "flex-end",
                          "backgroundColor": undefined,
                          "bottom": undefined,
                          "flex": 1,
                          "flexDirection": "column",
                          "flexGrow": undefined,
                          "gap": undefined,
                          "height": 108,
                          "justifyContent": "center",
                          "left": undefined,
                          "marginBottom": undefined,
                          "marginLeft": undefined,
                          "marginRight": undefined,
                          "marginTop": undefined,
                          "maxHeight": 108,
                          "maxWidth": undefined,
                          "minHeight": undefined,
                          "minWidth": 108,
                          "overflow": undefined,
                          "paddingBottom": undefined,
                          "paddingHorizontal": undefined,
                          "paddingLeft": undefined,
                          "paddingRight": undefined,
                          "paddingTop": undefined,
                          "paddingVertical": undefined,
                          "position": undefined,
                          "right": undefined,
                          "top": undefined,
                          "width": undefined,
                          "zIndex": undefined,
                        }
                      }
                      testID="header-right-pressable-test-id"
                    >
                      <View
                        onPress={[Function]}
                        style={
                          {
                            "alignItems": undefined,
                            "backgroundColor": undefined,
                            "bottom": undefined,
                            "flex": undefined,
                            "flexDirection": "column",
                            "flexGrow": undefined,
                            "gap": undefined,
                            "height": undefined,
                            "justifyContent": undefined,
                            "left": undefined,
                            "marginBottom": undefined,
                            "marginLeft": undefined,
                            "marginRight": undefined,
                            "marginTop": undefined,
                            "maxHeight": undefined,
                            "maxWidth": undefined,
                            "minHeight": undefined,
                            "minWidth": undefined,
                            "overflow": undefined,
                            "paddingBottom": undefined,
                            "paddingHorizontal": undefined,
                            "paddingLeft": undefined,
                            "paddingRight": undefined,
                            "paddingTop": undefined,
                            "paddingVertical": undefined,
                            "position": undefined,
                            "right": undefined,
                            "top": undefined,
                            "width": undefined,
                            "zIndex": undefined,
                          }
                        }
                        testID="header-right-icon-test-id"
                      >
                        <SvgMock
                          fill="transparent"
                          height={46}
                          stroke="rgba(255, 255, 255, 1)"
                          strokeLinecap="round"
                          strokeLinejoin="round"
                          strokeWidth={1}
                          width={46}
                        />
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
            <View
              style={
                {
                  "alignItems": undefined,
                  "backgroundColor": undefined,
                  "bottom": undefined,
                  "flex": undefined,
                  "flexDirection": "column",
                  "flexGrow": undefined,
                  "gap": undefined,
                  "height": undefined,
                  "justifyContent": undefined,
                  "left": undefined,
                  "marginBottom": undefined,
                  "marginLeft": undefined,
                  "marginRight": undefined,
                  "marginTop": undefined,
                  "maxHeight": undefined,
                  "maxWidth": undefined,
                  "minHeight": undefined,
                  "minWidth": undefined,
                  "overflow": undefined,
                  "paddingBottom": undefined,
                  "paddingHorizontal": undefined,
                  "paddingLeft": undefined,
                  "paddingRight": undefined,
                  "paddingTop": undefined,
                  "paddingVertical": undefined,
                  "position": undefined,
                  "right": undefined,
                  "top": undefined,
                  "width": undefined,
                  "zIndex": undefined,
                }
              }
            >
              <Text
                style={
                  {
                    "backgroundColor": undefined,
                    "bottom": undefined,
                    "color": "rgba(17, 17, 17, 1)",
                    "flexGrow": undefined,
                    "fontFamily": "Markpro-Medium",
                    "fontSize": 14,
                    "fontWeight": undefined,
                    "gap": undefined,
                    "height": undefined,
                    "left": undefined,
                    "letterSpacing": undefined,
                    "marginBottom": undefined,
                    "marginLeft": undefined,
                    "marginRight": undefined,
                    "marginTop": undefined,
                    "maxHeight": undefined,
                    "maxWidth": undefined,
                    "minHeight": undefined,
                    "minWidth": undefined,
                    "overflow": undefined,
                    "paddingBottom": undefined,
                    "paddingHorizontal": undefined,
                    "paddingLeft": undefined,
                    "paddingRight": undefined,
                    "paddingTop": undefined,
                    "paddingVertical": undefined,
                    "position": undefined,
                    "right": undefined,
                    "textAlign": "auto",
                    "top": undefined,
                    "width": undefined,
                    "zIndex": undefined,
                  }
                }
              >
                Test
              </Text>
            </View>
          </View>
        </RCTScrollView>
      </View>
    </View>
  </View>
</Modal>
`;
